server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -185,16 +182,6 @@ private PersistedState createOnDiskPersistedState(
         long lastAcceptedVersion = onDiskState.lastAcceptedVersion;
         long currentTerm = onDiskState.currentTerm;
 
-        if (onDiskState.empty()) {
```

Comment:
should we assert `onDiskState.empty() == false`?

Reply:
Unfortunately, we can have an empty cluster state on disk regardless whether we have on-disk global metadata files or not.